src/cli.ts

```diff
@@ -192,9 +192,9 @@ function transformMarkdown(buf: Buffer) {
   let state = 'root'
   let codeBlockEnd = ''
   let prevLineIsEmpty = true
-  const jsCodeBlock = /^(```+|~~~+)(js|javascript)$/
-  const shCodeBlock = /^(```+|~~~+)(sh|bash)$/
-  const otherCodeBlock = /^(```+|~~~+)(.*)$/
```

Comment:
The [problem](https://redosdetector.com/?pattern=%5E%28%60%60%60%2B%7C%7E%7E%7E%2B%29%28.*%29%24&caseInsensitive=false&unicode=false) here is that `.` won't match `\n` and a few other characters without the [`s` flag](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/dotAll).

Try running
```js
/^(```+|~~~+)(.*)$/.test('```' + '`'.repeat(1000000) + '\n')
```
in the console and it will lock up.

So the solution could be add the `s` flag, but actually given we weren't using the last group I think this makes more sense